packages/bootstrap-4/test/arrayFieldTemplate.test.tsx

```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import ArrayFieldTemplate from "../src/ArrayFieldTemplate";
+import DescriptionField from "../src/DescriptionField";
+import renderer from "react-test-renderer";
+import TitleField from "../src/TitleField";
+import { mockSchema } from "./helpers/createMocks";
+import { utils } from "@rjsf/core";
+const { getDefaultRegistry } = utils;
+
+describe("ArrayFieldTemplate", () => {
+  test("simple", () => {
+    const tree = renderer
+      .create(
+        <ArrayFieldTemplate
+          DescriptionField={() => <DescriptionField />}
+          TitleField={() => <TitleField />}
+          canAdd={true}
+          className=""
+          disabled={false}
+          onAddClick={() => void 0}
+          readonly={false}
+          required={false}
+          schema={mockSchema}
+          uiSchema={{}}
+          title=""
+          formContext={{}}
+          formData={{}}
+          registry={{ ...getDefaultRegistry() }}
+          // TODO : isSchema should be fixed here
+          // @ts-ignore
+          idSchema={{}}
```

Comment:
cc @jevakallio I think we need to solve this now or later ?

Reply:
Didn't we already figure out how to create the idSchema objects? Ideally we should fix it now since we're not going to roll back to do it.

If we don't know how to do this, let's just ping `@epicfaace` in the upstream PR and he can help us figure it out then.

Reply:
I actually forgot why I didnt use `Partial` here, 😄 

I can try to make it optional using `Partial`. But not sure then how the `core` will work

cause https://github.com/MLH-Fellowship/react-jsonschema-form/blob/ce4e79db192884bb075e1c08ff575ed99394ec94/packages/bootstrap-4/src/ArrayFieldTemplate/ArrayFieldTemplate.tsx#L41

I would change to 

```diff
- const id = `${idSchema.$id}__title`;
+ const id = `${idSchema && idSchema.$id}__title`; 
```

Reply:
I think that seems fine? I can't break anything, unless some calling code relies on it throwing an exception, which would be a terrible idea.